SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -449,6 +378,34 @@ void SpatialCommandUtils::TryKillProcessWithName(const FString& ProcessName)
 	}
 }
 
+void SpatialCommandUtils::TryGracefullyKill(FString ProcName, FProcHandle ProcHandle)
+{
+#if PLATFORM_WINDOWS
+	TryGracefullyKillWindows(ProcName);
+#else
+	// This sends a SIGTERM signal
+	FGenericPlatformProcess::TerminateProc(ProcHandle);
+#endif
+}
+
+void SpatialCommandUtils::TryGracefullyKillWindows(FString ProcName)
+{
+	// Use WM_CLOSE signal on windows as DestroyProc has forcefully kills when on Windows
+
+	// Find runtime window
+	const HWND RuntimeWindowHandle = FindWindowA(nullptr, TCHAR_TO_ANSI(*ProcName));
+	if (RuntimeWindowHandle != nullptr)
+	{
+		// Using SendMessageW to send message despite being 'supposed to' use SendMessage - as including "Windows/MinWindows.h" needed for
+		// SendMessage overrides the TEXT macro (SendMessageW is also used elsewhere in unreal)
+		SendMessageW(RuntimeWindowHandle, WM_CLOSE, NULL, NULL);
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Tried to gracefully stop process but could not find runtime window."));
```

Comment:
Might be cute to include the process name in the log.

Reply:
good point! added